src/Microsoft.AspNetCore.Mvc.TagHelpers/Internal/CacheTagHelperMemoryCacheFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Internal
+{
+    public class CacheTagHelperMemoryCacheFactory
+    {
+        public static readonly string MemoryCacheOptionsName = typeof(CacheTagHelper).FullName;
+
+        public CacheTagHelperMemoryCacheFactory(IOptionsMonitor<MemoryCacheOptions> optionsMonitor)
+        {
+            Cache = new MemoryCache(optionsMonitor.Get(MemoryCacheOptionsName));
+        }
+
+        // For testing only.
+        internal CacheTagHelperMemoryCacheFactory(IMemoryCache cache)
+        {
+            Cache = cache;
+        }
+
+        public IMemoryCache Cache { get; }
```

Comment:
Should we make this get;set to let people provide their own IMemoryCache implementation (not sure how common a scenario that would be, but I wouldn't think it's common).
Alternatively, just make the internal constructor public to allow people to register this type in DI with their own IMemoryCache instance.